### PR TITLE
Adding some check and add SCCs

### PR DESCRIPTION
**Pull request by matzew**
Adding some hack, for adding `scc` for a few "filtered" roles that do require it in order to run eventing on Openshift

The patched finalizers have been addressed upstream, hence I did not include them here.

But for our 0.5.0 we had to add the `scc`s to four cluster-roles, that are being massaged in this PR.

NOTE: we might need more roles for 0.6.0 to have these SCCs... (that's a tbd)

### CODE REVIEW

File: pkg/controller/install/install_controller.go [hunk 1]
```diff
@@ -161,3 +162,36 @@ func autoInstall(c client.Client, ns string) (err error) {
 	}
 	return err
 }
+
+func matchesClusterRole(roles []string, cr string) bool {
```
Comment (jcrossley3) [inline pkg/controller/install/install_controller.go]: It's a bit of a style thing, but I would prefer this function be local to `addSCC...`. See below.

File: pkg/controller/install/install_controller.go [hunk 2]
```diff
@@ -161,3 +162,36 @@ func autoInstall(c client.Client, ns string) (err error) {
...
+func addSCCforSpecialClusterRoles(u *unstructured.Unstructured) *unstructured.Unstructured {
+
+	// these do need some openshift specific SCC
+	clusterRoles := []string{
+		"eventing-broker-filter",
+		"knative-eventing-controller",
+		"in-memory-channel-controller",
+		"in-memory-channel-dispatcher",
+	}
+
```
Comment (jcrossley3) [inline pkg/controller/install/install_controller.go:175]: Given that Go provides no convenient `find` function and your roles are hard-coded anyway, I might move `matchesClusterRole` here where `clusterRoles` is in scope, e.g.
```suggestion
matchesClusterRole := func(cr string) bool {
	for _, i := range clusterRoles {
		if cr == i {
			return true
		}
	}
	return false
}
```

**Review by jcrossley3 — COMMENTED** [review]
My only nit is making the find function local to the only function that calls it.

**Comment by matzew** [comment]
OK, cool - will look at this later - thanks for the hint!

File: pkg/controller/install/install_controller.go [hunk 3]
```diff
@@ -161,3 +162,36 @@ func autoInstall(c client.Client, ns string) (err error) {
...
+		for _, i := range clusterRoles {
+			if cr == i {
+				return true
+			}
+		}
+		return false
+	}
+
+	// massage the roles that require SCC
+	if u.GetKind() == "ClusterRole" && matchesClusterRole(u.GetName()) {
```
Comment (matzew) [inline pkg/controller/install/install_controller.go:186]: @jcrossley3 I added the closure, to have the function local to `addSCCforSpecialClusterRoles` - thx for suggestion!

**Pull request merged**